packages/transaction-controller/src/TransactionController.ts

```diff
@@ -547,6 +561,21 @@ export class TransactionController extends BaseController<
       if (!this.isHistoryDisabled) {
         addInitialHistorySnapshot(transactionMeta);
       }
+
+      // Check if swaps transaction
+      if (
```

Comment:
Rather than added these conditions here, is it cleaner to always call `updateSwapsTransaction` but early return based on this same logic?

Reply:
Relocated in https://github.com/MetaMask/core/pull/1877/commits/517484795025534e391f60bb85dd706df3738c99